### ISSUE TITLE
Adding `dataTransfer` to `Html.Event`

### ISSUE THREAD

**Issue by pvdvreede** [mint-lang/mint#288]
Hi there, great project and am enjoying how simple it is to get started and up and running without needing to learn the plethora of node/js tools out there!

I am trying to implement some drag and drop using the native JS API (https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API) and noticed that the `Html.Event` does not have a `dataTransfer` property which is on a DragEvent (inherits from Mouse Event I think). 

How would you recommend storing state for that particular 'drag' event? Via native JS calls from the `Html.NativeEvent`? Could the `dataTransfer` property be added to Mint's Html.Event in the future?

Many thanks for your hard work on this project!

**Comment by gdotdesign** [comment]
> Hi there, great project and am enjoying how simple it is to get started and up and running without needing to learn the plethora of node/js tools out there!

Thanks :slightly_smiling_face: 

> How would you recommend storing state for that particular 'drag' event? Via native JS calls from the `Html.NativeEvent`? Could the `dataTransfer` property be added to Mint's Html.Event in the future?

You should use `Html.NativeEvent` for now, and in the future we will add to the `Html.Event` record.

**Comment by gdotdesign** [comment]
It's implemented in `master` will be released with `0.11.0`.

Here is an example of it:
```
component Main {
  fun handleDrop (event : Html.Event) {
    try {
      Html.Event.preventDefault(event)

      event.dataTransfer
      |> Html.DataTransfer.getData("text/plain")
      |> Debug.log()
    }
  }

  fun handleDragOver (event : Html.Event) {
    try {
      Html.Event.preventDefault(event)
      Html.DataTransfer.setDropEffect("link", event.dataTransfer)
    }
  }

  fun handleDragStart (event : Html.Event) {
    Html.DataTransfer.setData("text/plain", "Hello!", event.dataTransfer)
  }

  fun render {
    <div>
      <p
        onDrop={handleDrop}
        onDragOver={handleDragOver}>

        "Drop on ME!"

      </p>

      <p
        draggable="true"
        onDragStart={handleDragStart}>

        "This element is draggable."

      </p>
    </div>
  }
}
```
